src/main/java/com/purbon/search/fair/utils/MTableGenerator.java

```diff
@@ -6,37 +6,32 @@
 public class MTableGenerator {
 
     private int[] mTable;
-    @Deprecated
     private int n;
-    private int k;
     private double p;
     private double alpha;
 
-    public MTableGenerator(int k, double p, double alpha) {
-        this(k, k, p, alpha);
-    }
     /**
      * @param n     Total number of elements
-     * @param k     The size of the ranking
      * @param p     The proportion of protected candidates in the top-k ranking
      * @param alpha the significance level
+     * @param adjustAlpha
      */
-    @Deprecated
-    public MTableGenerator(int n, int k, double p, double alpha) {
-        if (parametersAreValid(n, k, p, alpha)) {
+    public MTableGenerator(int n, double p, double alpha, boolean adjustAlpha) {
```

Comment:
having boolean parameters like this one is a confusing thing. Can we have an options parameter (a map), where one of the options is going to be adjustAlpha.

we should also define a default value for adjust alpha.